src/building-blocks/FormFieldSet.tsx

```diff
@@ -31,22 +33,10 @@ class FormFieldSet extends Component<IFormFieldSetProps> {
     return fillInFieldSet(this.props.fieldSet);
   }
 
-  private get filteredFieldConfigs () {
-    // Filter out read-only fieldConfigs
-    const fieldConfigs = getFieldSetFields(this.fieldSet);
-
-    // If no fieldConfigs have an insertIf, skip costly formModel calculation
-    if (!fieldConfigs.some(fieldConfig => !!fieldConfig.insertIf)) {
-      return filterFieldConfigs(fieldConfigs, { readOnly: true });
-    }
-
-    // One of them does, so get formModel and filter out insertIfs
-    const formModel = this.props.formManager.formModel;
-    return filterFieldConfigs(fieldConfigs, { model: formModel, readOnly: true });
-  }
-
   public render () {
-    const filteredFieldConfigs = this.filteredFieldConfigs
+    const { formModel } = this.props
```

Comment:
Could all of this be done inside of the constructor or is it necessary to be computed the time of render? Doing this once at the beginning of the lifecycle would likely also have significant performance improvements if it's possible

Reply:
Yeah but insertIf takes the form model as an input and can ( and usually does ) return different results when the form model changes.